### PR TITLE
Update references to documentation projects on the main branch

### PR DESCRIPTION
**Pull request by cachafla**
## Internal Notes for Reviewers

- Brought all documentation changes where we updated all references to "documentation projects" for "model" since we no longer expose "projects" as a concept on the UI.
- Fix validmind/tests/data_validation/TabularCategoricalBarPlots.py
- Fix issue where inputs were not being tracked where the input value was a list of inputs

### CODE REVIEW

File: notebooks/code_sharing/clustering/quickstart_cluster_demo.ipynb
Comment (AnilSorathiya) [inline notebooks/code_sharing/clustering/quickstart_cluster_demo.ipynb:1]: we don't have separate clustering usecase template available in the backend repo yet. Can we create one?

File: notebooks/code_sharing/insurance_mortality/insurance_validation_demo.ipynb
Comment (AnilSorathiya) [inline notebooks/code_sharing/insurance_mortality/insurance_validation_demo.ipynb:1]: We might need to revisit this file later to define purpose of it. We are suggesting to use binary classification template but then we are running regression tests that are not listed in the template.

**Review by AnilSorathiya — APPROVED** [review]
Looking good to me.  👍 
I have commented at a couple of places to create new templates for making visible at UI.

File: notebooks/code_sharing/clustering/quickstart_cluster_demo.ipynb
Comment (cachafla) [reply notebooks/code_sharing/clustering/quickstart_cluster_demo.ipynb:1]: Makes sense. We. need to build a simple cluster use case template.

File: notebooks/code_sharing/insurance_mortality/insurance_validation_demo.ipynb
Comment (cachafla) [reply notebooks/code_sharing/insurance_mortality/insurance_validation_demo.ipynb:1]: Yeah. We might also delete this file since it's too old.

**Pull request merged**